packages/flutter_tools/lib/src/doctor.dart

```diff
@@ -148,44 +162,62 @@ class Doctor {
 
   /// Print a summary of the state of the tooling, as well as how to get more info.
   Future<void> summary() async {
-    printStatus(await summaryText);
+    printStatus(await _summaryText());
   }
 
-  Future<String> get summaryText async {
+  Future<String> _summaryText() async {
     final StringBuffer buffer = StringBuffer();
 
-    bool allGood = true;
+    bool missingComponent = false;
+    bool sawACrash = false;
 
     for (DoctorValidator validator in validators) {
       final StringBuffer lineBuffer = StringBuffer();
-      final ValidationResult result = await validator.validate();
-      lineBuffer.write('${result.coloredLeadingBox} ${validator.title} is ');
+      ValidationResult result;
+      try {
+        result = await asyncGuard<ValidationResult>(() => validator.validate());
+      } catch (exception) {
+        // We're generating a summary, so drop the stack trace.
+        result = ValidationResult.crash(exception);
+      }
+      lineBuffer.write('${result.coloredLeadingBox} ${validator.title}: ');
       switch (result.type) {
+        case ValidationType.crash:
+          lineBuffer.write('the doctor check crashed without a result.');
+          sawACrash = true;
+          break;
         case ValidationType.missing:
-          lineBuffer.write('not installed.');
+          lineBuffer.write('is not installed.');
           break;
         case ValidationType.partial:
-          lineBuffer.write('partially installed; more components are available.');
+          lineBuffer.write('is partially installed; more components are available.');
           break;
         case ValidationType.notAvailable:
-          lineBuffer.write('not available.');
+          lineBuffer.write('is not available.');
           break;
         case ValidationType.installed:
-          lineBuffer.write('fully installed.');
+          lineBuffer.write('is fully installed.');
           break;
       }
 
-      if (result.statusInfo != null)
+      if (result.statusInfo != null) {
         lineBuffer.write(' (${result.statusInfo})');
+      }
 
       buffer.write(wrapText(lineBuffer.toString(), hangingIndent: result.leadingBox.length + 1));
       buffer.writeln();
 
-      if (result.type != ValidationType.installed)
-        allGood = false;
+      if (result.type != ValidationType.installed) {
+        missingComponent = false;
+      }
+    }
+
+    if (sawACrash) {
+      buffer.writeln();
+      buffer.writeln('Run "flutter doctor" for information about why a doctor check crashed.');
```

Comment:
in verbose mode?

Reply:
W.r.t. crashes the difference between -v and no -v is that the stacktrace is included. I didn't want to steer folks towards that initially.